relayer/processor/path_processor_internal.go

```diff
@@ -618,7 +634,91 @@ func (pp *PathProcessor) assembleAndSendMessages(
 		outgoingMessages = append(outgoingMessages, message)
 	}
 
-	// TODO handle connection and channel handshake messages
+	for _, msg := range messages.connectionMessages {
```

Comment:
This function would become more readable if the `for .. range ..` blocks were inside their own functions.

Reply:
good call, done!